bayes_opt/bayesian_optimization.py

```
@@ -442,3 +446,233 @@ def load_state(self, path: str | PathLike[str]) -> None:
+    warmup_steps: int, optional (default=500)
+        Number of warmup steps for MCMC sampling.
+    thinning: int, optional (default=16)
+        Thinning factor for MCMC samples to reduce autocorrelation.
+    """
+
+    def __init__(
+        self,
+        f: Callable[..., float] | None,
+        pbounds: Mapping[str, tuple[float, float]],
+        acquisition_function: AcquisitionFunction | None = None,
+        constraint: Optional[NonlinearConstraint] = None,
```

Ruff (0.11.9)

Undefined name `Optional`

(F821)